### ISSUE TITLE
Mimic `base::read.table()` variable name fixing

### ISSUE THREAD

**Issue by wibeasley** [hadley/readr#165]
Consider if `readr::read_xxx()` should have a parameter (possibly `make_names=F`) that specifies if unconventional variable names should be renamed in the same way as `base::read.xxx()` does.  Admittedly it's minor, but it would allow the readr functions to be drop-in compatible in more code.    

My guess is that compatibility could be accomplished by calling [`base::make.names()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/make.names.html) somewhere near the bottom of [`readr::read_delimited()`](https://github.com/hadley/readr/blob/master/R/read_delim.R)

As a small example, I'm working with code that needed to change from
```r
ds <- plyr::rename(ds, replace=c(
    "Subject." = "subject_id"
  , "Art." = "ff_art" 
))
```
to
```r
ds <- plyr::rename(ds, replace=c(
   "Subject#" = "subject_id"
  , "Art " = "ff_art" #Be careful of the trailing space after 'Art '
))
```

In other words, somewhere `readr::read_delimiter would do something like this, and copy over the data.frame colnames.

```r
> make.names(c("Subject#", "Art "))
[1] "Subject." "Art."  
```
We've certainly all been given datasets with many more violating variables than two.

**Comment by hadley** [comment]
This is by design.  I think changing column names is a bad idea, especially since they often contain data.

**Comment by wibeasley** [comment]
Because it's potentially a many-to-one transformation? (ie `Subject?` and `Subject#` both resolve to `Subject.`?)

Cool.
